pkgs/build-support/setup-hooks/mpi-check-hook/mpi-check-hook.sh

```diff
@@ -0,0 +1,50 @@
+preCheckHooks+=('setupMpiCheck')
+preInstallCheckHooks+=('setupMpiCheck')
+
+
+setupMpiCheck() {
+  # Find out which MPI implementation we are using
+  # and set safe defaults that are guaranteed to run
+  # on any build machine
+
+  mpiType="NONE"
+
+  # OpenMPI signature
+  if command ompi_info &> /dev/null; then
+    mpiType="openmpi"
+  fi
+
+  # MPICH based implementations
+  if command mpichversion &> /dev/null; then
+    version=$(mpichversion)
+    if [[ "$version" == *"MPICH"* ]]; then
+      mpiType="MPICH"
+    fi
+    if [[ "$version" == *"MVAPICH"* ]]; then
+      mpiType="MVAPICH"
+    fi
+  fi
```

Comment:
In the unlikely event of two MPI implementations being present, this will be ambiguous. Maybe add a warning or error for such a case?

Reply:
In the build sandbox this is rather unlikely, but it could happen in a nix-shell.